src/api.ts

```diff
@@ -340,10 +340,14 @@ export class SharedCache {
     return SharedCache.instance;
   }
 
-  public getPromptCacheKey(id: string, name: string, version: number): string {
+  public getPromptCacheKey(
+    id?: string,
+    name?: string,
+    version?: number
+  ): string {
     if (id) {
       return id;
-    } else if (name && version) {
+    } else if (name && (version || version === 0)) {
```

Comment:
There is actually a way to do that explicitly:
```suggestion
    } else if (name && typeof version === "number") {
```